RevenueCat/Plugins/iOS/PurchasesUnityHelper.m

```diff
@@ -442,6 +445,114 @@ - (void)showInAppMessages:(NSArray<NSNumber*>*)messageTypes {
     #endif
 }
 
+- (void)getEligibleWinBackOffersForProduct:(NSString *)productIdentifier {
+    // Checking for win-back offer APIs being available in the current OS is handled at the PHC level.
+    [RCCommonFunctionality eligibleWinBackOffersForProductIdentifier:productIdentifier
+                                                        completionBlock:^(NSArray<NSDictionary *> * _Nullable eligibleWinBackOffers, RCErrorContainer * _Nullable errorContainer) {
+
+        RCErrorContainer *finalErrorContainer = errorContainer;
+        if (eligibleWinBackOffers == nil && errorContainer == nil) {
+            NSError *nsError = [[NSError alloc] initWithDomain:RCPurchasesErrorCodeDomain
+                                                            code:RCUnknownError
+                                                        userInfo:@{NSLocalizedDescriptionKey: @"Both error and response are nil"}];
+            finalErrorContainer = [[RCErrorContainer alloc] initWithError:nsError extraPayload:@{}];
+        }
+
+        // Send error if present
+        if (finalErrorContainer != nil) {
+            NSDictionary *response = @{@"error": finalErrorContainer.info};
+            [self sendJSONObject:response toMethod:GET_ELIGIBLE_WIN_BACK_OFFERS_FOR_PRODUCT];
+            return;
+        }
+
+        // Send response dictionary if present
+        NSArray *offers = eligibleWinBackOffers ?: @[];
+        NSDictionary *response = @{@"eligibleWinBackOffers": offers};
+        [self sendJSONObject:response toMethod:GET_ELIGIBLE_WIN_BACK_OFFERS_FOR_PRODUCT];
+    }];
+}
+
+// This function accepts a product identifier since the PHC code only fetches eligible win-back offers for products
+- (void)getEligibleWinBackOffersForPackage:(NSString *)productIdentifier {
```

Comment:
Hmm I'm wondering if we need this method at all, and we should just have `getEligibleWinBackOffersForProduct`, and have the C# side only call that one internally... Wdyt?

Reply:
Yeah, we can probably get away with just `getEligibleWinBackOffersForProduct`. I'll work on making that change 💪

Reply:
Ah, I remember why I kept them separate now 😅

The native code here in `PurchasesUnityHelper.m` determines which function it sends the JSON response to:

```objc
[self sendJSONObject:response toMethod:GET_ELIGIBLE_WIN_BACK_OFFERS_FOR_PRODUCT];
```

I think the only way this would work is if `PurchasesUnityHelper.getEligibleWinBackOffersForProduct` took in the method name as an argument, which clouds up the API there instead. I felt like it was clearer to keep them as separate functions instead of adding that extra parameter there

Reply:
Ahh that makes sense yeah... I guess we could try to add another level of indirection in the C# side to avoid that... But that's going to complicate things and not follow the current architecture, so I think this is better. Thanks 🙏